### PR TITLE
feat: added import mode

### PR DESCRIPTION
**Pull request by gumuz**
fixes https://github.com/GetStream/chat/issues/3502

### CODE REVIEW

**Review by yaziine — APPROVED** [review]

File: src/client.ts [hunk 1]
```diff
@@ -2860,12 +2860,13 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
    *
    * @private
    * @param {string} path path of uploaded data
-   *
+   * @param {string} mode import mode insert|upsert
    * @return {APIResponse & CreateImportResponse} An ImportTask
    */
-  async _createImport(path: string) {
+  async _createImport(path: string, mode: string) {
```
Comment (ferhatelmas) [inline src/client.ts:2866]: it might wiser to take second param as an option object so that we can add more flags in the future if needed

Comment (ferhatelmas) [reply src/client.ts:2866]: since it's noted to be changed, not strict though

File: src/client.ts [hunk 2]
```diff
@@ -2864,12 +2865,13 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
    *
    * @private
    * @param {string} path path of uploaded data
-   *
+   * @param {CreateImportOptions} options import options
    * @return {APIResponse & CreateImportResponse} An ImportTask
    */
-  async _createImport(path: string) {
+  async _createImport(path: string, options: CreateImportOptions) {
```
Comment (ferhatelmas) [inline src/client.ts:2871]: accept default too

Comment (ferhatelmas) [reply src/client.ts:2871]: `_createImport(path)` should be fine

File: src/client.ts [hunk 3]
```diff
@@ -624,18 +625,18 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
   }
 
   /**
-	 * testPushSettings - Tests the push settings for a user with a random chat message and the configured push templates
```
Comment (gumuz) [inline src/client.ts:627]: not sure why linter didn't catch this before @ferhatelmas

Comment (ferhatelmas) [reply src/client.ts:627]: yes interesting

File: src/client.ts [hunk 4]
```diff
@@ -2866,12 +2867,13 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
    *
    * @private
    * @param {string} path path of uploaded data
-   *
+   * @param {CreateImportOptions} options import options
    * @return {APIResponse & CreateImportResponse} An ImportTask
    */
-  async _createImport(path: string) {
+  async _createImport(path: string, options: CreateImportOptions = { mode: 'upsert' }) {
```
Comment (ferhatelmas) [inline src/client.ts:2873]: this is good, but we could treat missing as upsert probably

**Review by ferhatelmas — APPROVED** [review]

**Review by ferhatelmas — APPROVED** [review]

**Pull request closed without merging**